### PR TITLE
Update docs for testnet3

### LINKED ISSUE #710: Testnet Documentation suggestions
1. Write separate guides for docker & rust

> image

2. https://lighthouse-book.sigmaprime.io/docker.html 
Add next steps to continue with docker 

>docker run -it --name=light-beacon -v $HOME/lighthouse-data:/root/.lighthouse -p >127.0.0.1:10000:10000 lighthouse lighthouse beacon_node --http-address=0.0.0.0 --http-port=10000 ->-http

>docker run --name=light-valid-keygen -v $HOME/lighthouse-data:/root/.lighthouse lighthouse >lighthouse account validator new random

>docker run -it --name=light-valid -v $HOME/lighthouse-data:/root/.lighthouse --network="host" >lighthouse lighthouse validator --server http://0.0.0.0:10000/ 

@agemanning

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

Resolves #710

## Proposed Changes

Updates the testnet docs with:
- Separate docs for `docker-compose` and building from source.
- Info about ports and volumes for the docker section.

### CODE REVIEW

File: book/src/docker.md [hunk 1]
```diff
@@ -3,17 +3,67 @@
 This repository has a `Dockerfile` in the root which builds an image with the
 `lighthouse` binary installed.
 
-To use the image, first build it (this will likely take several minutes):
+A pre-built image is available on Docker Hub and the
+[sigp/lighthouse](https://github.com/sigp/lighthouse-docker) repository
+contains a full-featured `docker-compose` environment.
+
+## Obtaining the Docker image
```
Comment (pscott) [inline book/src/docker.md]: I would add a comment here specifying that you can either use Docker Hub OR build the docker image from source.
e.g. "There are two ways of obtaining the docker image, either via Docker Hub or build the Docker Image from source. Once you have the docker image, proceed to [using the docker image](https://lighthouse-book.sigmaprime.io/docker.html#Building-the-Docker-Image)" ?

File: book/src/become-a-validator-docker.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+[sigp/lighthouse-docker](https://github.com/sigp/lighthouse-docker) repository.
+
+```bash
+$ git clone https://github.com/sigp/lighthouse-docker
+$ cd lighthouse-docker
+```
+
+### 2. Configure the docker environment
+
+Then, create a `.env` file with the following contents (these values are
```
Comment (pscott) [inline book/src/become-a-validator-docker.md]: I would suggest being more explicit (had a misunderstanding while following the guide):
`Then, create a file named `.env` with the following contents  ...`

File: book/src/become-a-validator-source.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+**TODO: document starting an eth1 node**
+
+## 1. Download and install Lighthouse
+
+If you already have Rust installed, you can install Lighthouse with the
+following three commands:
+
+- `$ git clone https://github.com/sigp/lighthouse.git`
+- `$ cd lighthouse`
+- `$ make`
```
[the remark this answers is not in the document]
Comment (paulhauner) [reply book/src/become-a-validator-source.md]: I've linked to the the troubleshooting section, but I haven't added those deps as I'm not sure which distros they link to.

Feel free to make a PR and update :)

[the remark this answers is not in the document]
Comment (paulhauner) [reply book/src/become-a-validator-source.md]: Oh also, `ganache-cli` is mentioned on the "Development Environment" page.

File: book/src/become-a-validator-source.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+help menu.
+
+> - If you're not familiar with Rust or you'd like more detailed instructions,
+>   see the [Installation Guide](./installation.md) which contains a
+>   [Troubleshooting](installation.html#troubleshooting) section.
+
+## 2. Start an Eth1 client
+
+As Eth2 relies upon the Eth1 chain for validator on-boarding and eventually
+Eth1 may use the Eth2 chain as a finality gadget. As such, all Eth2 validators
```
Comment (pscott) [inline book/src/become-a-validator-source.md]: I believe there's a wording error here? "As [...].  As such [..]"?
Maybe I'm completely wrong here, but I think remove ". As such, " should suffice.
"As Eth2 relies upon the Eth1 chain for validator on-boarding and eventually
Eth1 may use the Eth2 chain as a finality gadget, all Eth2 validators ..."

Comment (paulhauner) [reply book/src/become-a-validator-source.md]: Thank you!

**Pull request merged**